tokio/src/stream/mod.rs

```diff
@@ -653,6 +659,66 @@ pub trait StreamExt: Stream {
     {
         Collect::new(self)
     }
+
+    /// Applies a per-item timeout to the passed stream.
+    ///
+    /// `timeout()` takes a `Duration` that represents the maximum amount of
+    /// time each element of the stream has to complete before timing out.
+    ///
+    /// If the wrapped stream yields a value before the deadline is reached, the
+    /// value is returned. Otherwise, an error is returned.
```

Comment:
We probably want to add a note specifying that the caller *may* decide to continue consuming the stream and will eventually get the next source stream value once it becomes available.

My only question regarding behavior is whether or not we should guarantee that only *one* elapsed error will be triggered between source stream values or there can be one per elapsed interval.

Reply:
We probably want to add an example that does something like:

```rust
my_stream
    .timeout(Duration::from_millis(100))
    .take_while(Result::is_ok)
```

Reply:
Added the comment as you suggested. There was already an example showing how to use the `take_while` operator as part of the larger example. It is not set to 'really' run though because  it is very flaky.

If you prefer, I can separate the take_while usage into a separate example and have it actually run.